iOS/UI/Swipeable Input/SwipeableInputAccessoryView.swift

```diff
@@ -126,64 +147,148 @@ class SwipeableInputAccessoryView: BaseView, UIGestureRecognizerDelegate, Active
+
+    private func handleInputTap() {
+        if self.textView.isFirstResponder {
+            // When the text view is editing, double taps should expand it.
+            self.inputState = .expanded
+        } else {
+            // If we're not editing, a tap starts editing.
+            self.textView.updateInputView(type: .keyboard, becomeFirstResponder: true)
+        }
+    }
+
+    private func handleBackgroundTap() {
+        if self.inputState == .expanded {
+            self.inputState = .collapsed
+        } else if self.textView.isFirstResponder {
+            self.textView.resignFirstResponder()
+        }
+    }
 
     private func setupHandlers() {
         self.updateInputType(with: .keyboard)
         
         KeyboardManager.shared
             .$currentEvent
-            .mainSink { [weak self] currentEvent in
-                guard let `self` = self else { return }
-                
+            .mainSink { [unowned self] currentEvent in
                 switch currentEvent {
                 case .willShow:
                     self.showDetail(shouldShow: true)
+                    self.hintAnimator.updateSwipeHint(shouldPlay: false)
                 case .willHide:
                     self.showDetail(shouldShow: false)
+                    self.hintAnimator.updateSwipeHint(shouldPlay: false)
                 case .didHide:
                     self.textView.updateInputView(type: .keyboard, becomeFirstResponder: false)
+                    self.inputState = .collapsed
                 default:
                     break
                 }
-                self.updateSwipeHint(shouldPlay: false)
             }.store(in: &self.cancellables)
         
         self.textView.$inputText.mainSink { [unowned self] text in
             self.handleTextChange(text)
-            self.updateHeight(with: self.textView.numberOfLines)
+            self.updateInputState(with: self.textView.numberOfLines)
             self.countView.update(with: text.count, max: self.textView.maxLength)
         }.store(in: &self.cancellables)
-        
-        self.overlayButton.didSelect { [unowned self] in
-            if !self.textView.isFirstResponder {
-                self.textView.updateInputView(type: .keyboard, becomeFirstResponder: true)
-            }
-        }
-        
-        self.textView.confirmationView.button.didSelect { [unowned self] in
-            self.didPressAlertCancel()
-        }
+
+        self.textView.$isEditing.mainSink { [unowned self] isEditing in
+            // If we are editing, a double tap should trigger the expanded state.
+            // If we're not editing, it takes 1 tap to start.
+            self.inputFieldTapRecognizer.numberOfTapsRequired = isEditing ? 2 : 1
+        }.store(in: &self.cancellables)
+
+        self.$inputState
+            .removeDuplicates()
+            .mainSink { [unowned self] inputState in
+                self.updateLayout(for: inputState)
+            }.store(in: &self.cancellables)
```

Comment:
Publishers are nice. Whenever the input state changes, we guarantee the update layout function is called.